firefox_puppeteer/tests/test_tabs.py

```diff
@@ -2,78 +2,152 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from marionette import Wait
-from marionette.errors import NoSuchElementException
-
 from firefox_ui_harness.testcase import FirefoxTestCase
 
 
-class TestTabs(FirefoxTestCase):
+class TestTabBar(FirefoxTestCase):
+
+    def test_basics(self):
+        tabbar = self.browser.tabbar
+
+        self.assertEqual(tabbar.window, self.browser)
```

Comment:
I think this reveals a problem with the overall strategy of attaching generic functionality to a window object and pushing the reference through to subclasses. This relationship is circular, and makes the idea of a "window" in our code as reflecting a ui element less meaningful.